File: python/lengthOfLastWord.py
# 比较精简的解法
class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        s.rstrip()
        if(len(s) == 1):
            return 1
        s_new = s.split(" ")
        return len(s_new[-1])

# 下面这种方法可能会更快一些
# class Solution(object):
#     def lengthOfLastWord(self, s):
#         """
#         :type s: str
#         :rtype: int
#         """
#         if s.isspace():
#             return 0
#         length = len(s)
#         j = 0
#         ret = 0
#         for i in range(length-1,0,-1):
#             if s[i] == ' ':
#                 pass
#             else:
#                 j = i
#                 break
#         for i in range(j,0,-1):
#             if(s[i] == ' '):
#                 break
#             else:
#                 ret++
#         return ret

File: c++/lengthOfLastWord.cc
#include <string>
#include <ctype.h>

using namespace std;

class Solution {
public:
    int lengthOfLastWord(string s) {
        int ret = 0;
        if(s.length() == 0)
            return 0;
        int i;
        for(i = s.length()-1; i>=0; --i) {
            if(!isblank(s[i]))
                break;
        }
        if(i == 0 && isblank(s[i]))
            return 0;
        for(int j = i; j>=0; --j) {
            if(isblank(s[j]))
                break;
            ++ret;
        }
        return ret;
    }
};

